src/OmniSharp.Roslyn.CSharp/Services/QuickInfoProvider.cs

```diff
@@ -0,0 +1,197 @@
+﻿using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.QuickInfo;
+using Microsoft.CodeAnalysis.Text;
+using OmniSharp.Mef;
+using OmniSharp.Models.v2;
+using OmniSharp.Options;
+
+#nullable enable
+
+namespace OmniSharp.Roslyn.CSharp.Services
+{
+    [OmniSharpHandler(OmniSharpEndpoints.V2.QuickInfo, LanguageNames.CSharp)]
+    public class QuickInfoProvider : IRequestHandler<QuickInfoRequest, QuickInfoResponse>
+    {
+        // Based on https://github.com/dotnet/roslyn/blob/master/src/Features/LanguageServer/Protocol/Handler/Hover/HoverHandler.cs
```

Comment:
maybe could you link to a commit instead of head since the head link might break when the Roslyn code is rearranged

Reply:
ProTip: If you go to the url in github and hit `y` on your keyboard it will "Expand URL to its canonical form" (you can see the shortcut in the github help menu `?`)

Reply:
Well, it does when you don't have vimium enabled :)